File: sample_daggerandroid/src/main/java/io/github/hurshi/sample_daggerandroid/SecActivity.java
package io.github.hurshi.sample_daggerandroid;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.FragmentTransaction;
import android.util.Log;

import javax.inject.Inject;

import dagger.android.support.DaggerAppCompatActivity;
import io.github.hurshi.sample_daggerandroid.beans.AppBean;
import io.github.hurshi.sample_daggerandroid.beans.SecActivityBean;
import io.github.hurshi.sample_daggerandroid.modules.SecActivityModule;
import io.github.hurshi.sample_daggerandroid.scopes.ActivityScope;
import io.github.hurshi.simplifydagger.annotation.AutoAndroidComponent;


@AutoAndroidComponent(scope = ActivityScope.class, modules = {SecActivityModule.class}, fragments = SecFragment.class)
public class SecActivity extends DaggerAppCompatActivity {
    @Inject
    SecActivityBean secActivityBean;

    @Inject
    AppBean appBean;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_sec);
        Log.e(">>>", "log from SecActivity  " + appBean.toString());
        Log.e(">>>", "log from SecActivity  " + secActivityBean.toString());

        loadFragment();
    }

    private void loadFragment() {
        SecFragment fragment = new SecFragment();
        FragmentTransaction transaction = getSupportFragmentManager().beginTransaction();
        transaction.add(R.id.frame_layout, fragment);
        transaction.commit();
    }
}

File: app/src/main/java/io/github/hurshi/simplifydagger/SecActivity.kt
package io.github.hurshi.simplifydagger

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.util.Log
import com.google.gson.Gson
import io.github.hurshi.simplifydagger.annotation.AutoComponent
import io.github.hurshi.simplifydagger.beans.Teacher
import io.github.hurshi.simplifydagger.modules.SecModule
import io.github.hurshi.simplifydagger.scopes.ActivityScope
import javax.inject.Inject

@AutoComponent(module = SecModule::class, scope = ActivityScope::class)
class SecActivity : AppCompatActivity() {
    @Inject
    lateinit var teacher: Teacher

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        DaggerAutoSecActivityComponent.create().inject(this)
        Log.e(">>>", "teacher = ${Gson().toJson(teacher)}")
    }
}